x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/action/privilege/PutPrivilegesResponseTests.java

```diff
@@ -21,7 +21,6 @@
 
 public class PutPrivilegesResponseTests extends ESTestCase {
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/82831")
     public void testSerialization() throws IOException {
         final int applicationCount = randomInt(3);
         final Map<String, List<String>> map = Maps.newMapWithExpectedSize(applicationCount);
```

Comment:
Isn't the map creation in the test itself still problematic since the response is going to be in natural order but here we're still using a hashmap?

If there's no expectation on ordering here I really think the _better_ thing to do is make the test more robust and indifferent to ordering rather than have to have to use stable collections all over the place.

Reply:
Good point @mark-vieira! I've reverted the change in `PutPrivilegesResponse` and amended the assertions to check the JSONs not as strings, but as maps.